web-ui/src/main/resources/catalog/components/viewer/projectionSwitcher/ProjectionSwitcher.js

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2001-2016 Food and Agriculture Organization of the
+ * United Nations (FAO-UN), United Nations World Food Programme (WFP)
+ * and United Nations Environment Programme (UNEP)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or (at
+ * your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ * Contact: Jeroen Ticheler - FAO - Viale delle Terme di Caracalla 2,
+ * Rome - Italy. email: geonetwork@osgeo.org
+ */
+
+(function() {
+  goog.provide('gn_projectionswitcher');
+
+  var module = angular.module('gn_projectionswitcher', [
+  ]);
+
+
+  /**
+   * @ngdoc directive
+   * @name gn_projectionswitcher.directive:gnProjectionSwitcher
+   * 
+   * @description The `gnProjectionSwitcher` directive adds a button to the map
+   *              that allows you to change the projection.
+   */
+  module.directive('gnProjectionSwitcher', [
+    'gnViewerSettings',
+    function(gnViewerSettings) {
+      return {
+        restrict: 'A',
+        templateUrl: '../../catalog/components/viewer/projectionSwitcher/' +
+            'partials/projectionSwitcher.html',
+        scope: {
+          map: '=gnProjectionSwitcher'
+        },
+        controllerAs: 'gnProjectionSwitcherCtrl',
+        controller: ['$scope', 'gnViewerSettings', 'gnMap', '$rootScope', 
+          function(scope, gnViewerSettings, gnMap, $rootScope) {
+
+          scope.changeLayerProjection = function(
+            layer, oldProj, newProj) {
+            if (layer instanceof ol.layer.Group) {
+              layer.getLayers().forEach(
+                  function(subLayer) {
+                    this.changeLayerProjection(
+                      subLayer, oldProj,
+                      newProj);
+                  });
+            } else if (layer instanceof ol.layer.Tile) {
+              var tileLoadFunc = layer.getSource()
+                .getTileLoadFunction();
+              layer.getSource().setTileLoadFunction(
+                tileLoadFunc);
```

Comment:
Does this actually work? I would have thought you needed to create a new Source object with the new projection, and only if the source actually handles the new projection.